### PR TITLE
Improvements windows installer script

### LINKED ISSUE #34: [Windows] Install script mixes up user and machine PATH definitions
When the script runs in non-privileged mode, we set the User's PATH but fill it with the amended Machine path.

```ps1
  [Environment]::SetEnvironmentVariable(
      "Path",
      [Environment]::GetEnvironmentVariable("Path", [EnvironmentVariableTarget]::Machine) + ";$INSTALL_FOLDER",
      [EnvironmentVariableTarget]::User)
```

### PR DESCRIPTION
**Pull request by pysco68**
- cleaned up messages
- fixed #34
- "fixed" #31 by testing writability and outputting appropriate message to the user
- Added detection for the tipi install folder already being on PATH
- Improved and unified a bunch of messages
- 
fix nxxm/nxxm-src#674

### CODE REVIEW

**Comment by pysco68** [comment]
⚠️ We have to switch the cloud installation to system install before we proceed merging this one ⚠️ 

Otherwise tipi will be installed as user "packer" during installation and won't be available to our user in the end.

File: install/install_for_macos_linux.sh [hunk 1]
```diff
@@ -47,9 +47,11 @@ sudo unzip ~/tipi.zip -d $INSTALL_FOLDER -x LICENSE && rm ~/tipi.zip
 if [ $? -eq 0 ]; then
     info "tipi successfully installed. Installing the dependencies..."
     sudo chmod +x $INSTALL_FOLDER/bin/tipi
-    $INSTALL_FOLDER/bin/tipi --help --dont-upgrade
+    $INSTALL_FOLDER/bin/tipi --dont-upgrade run echo "Done"
```
Comment (daminetreg) [inline install/install_for_macos_linux.sh:50]: THese changes are also #36. but they are compatible

File: install/install_for_windows.ps1 [hunk 2]
```diff
@@ -1,90 +1,147 @@
...
+    $INSTALL_FOLDER = Join-Path -Path ([Environment]::GetFolderPath('LocalApplicationData')) -ChildPath "\tipi"
+}
+
+if($system_install) {
+    $INSTALL_FOLDER = "C:\ProgramData\tipi"
 }
-$INSTALL_FOLDER="C:\ProgramData\tipi"
-$TIPI_EXE="$INSTALL_FOLDER\tipi.exe"
-$TIPI_URL="https://github.com/tipi-build/cli/releases/download/$version_to_use/tipi-$version_to_use-windows-win64.zip"
+
```
Comment (daminetreg) [inline install/install_for_windows.ps1:22]: If I set TIPI_INSTALL_DEST + TIPI_INSTALL_SYSTEM I end up installed in ProgramData nonetheless.

**Review by daminetreg — CHANGES_REQUESTED** [review]
Great. I'm afraid of the combinatory but looks fine.

- [x] TIPI_INSTALL_DEST not respected in all cases

Regarding the TIPI_INSTALL_DEST that isn't respect when TIPI_INSTALL_SYSTEM is wanted I thinks it's a problem, and I believe can be in the remote environment provisioning.

**Review by daminetreg — APPROVED** [review]

**Review by daminetreg — APPROVED** [review]

**Pull request merged**